pkg/storage/storage.go

```diff
@@ -145,3 +152,98 @@ func varyVoter(baseKey string, req *http.Request, currentKey string) bool {
 
 	return false
 }
+
+func mappingElection(provider types.Storer, item []byte, req *http.Request, validator *rfc.Revalidator, logger *zap.Logger) (resultFresh *http.Response, resultStale *http.Response, e error) {
+	var mapping types.StorageMapper
+	if len(item) != 0 {
+		e = json.NewDecoder(bytes.NewBuffer(item)).Decode(&mapping)
```

Comment:
JSON? Wouldn't we want a binary encoder here instead? (since it's an internal map)

I will test performances and memory usage. When inserting millions of keys, there might be a visible difference.